gcloud/datastore/query.py

```diff
@@ -84,26 +94,12 @@ def __init__(self,
         self._group_by = list(group_by)
 
     @property
-    def dataset(self):
-        """Get the dataset for this Query.
-
-        The dataset against which the Query will be run.
-
-        :rtype: :class:`gcloud.datastore.dataset.Dataset` or None,
-        :returns: the current dataset.
-        """
-        return self._dataset
-
-    @dataset.setter
-    def dataset(self, value):
-        """Set the dataset for the query
+    def dataset_id(self):
```

Comment:
Yeah I have the same intuition, just wanted to play the other side of things.

It seems allowing `dataset_id` to change on a query could result in user confusion.